src/app/Dashboard/Charts/ChartCard.tsx

```diff
@@ -0,0 +1,367 @@
+  }, [addSubscription, controllerContext, setHasRecording]);
+
+  React.useEffect(() => {
+    addSubscription(serviceContext.api.grafanaDashboardUrl().subscribe(setDashboardUrl));
+  }, [addSubscription, serviceContext, setDashboardUrl]);
+
+  React.useEffect(() => {
+    if (!dashboardUrl) {
+      return;
+    }
+    const u = new URL('/d-solo/main', dashboardUrl);
+    u.searchParams.append('theme', props.theme);
+    u.searchParams.append('panelId', String(kindToId(props.chartKind)));
+    u.searchParams.append('to', 'now');
+    u.searchParams.append('from', `now-${props.duration}s`);
+    u.searchParams.append('refresh', `${props.period}s`);
+    setChartSrc(u.toString());
+  }, [dashboardUrl, props.theme, props.chartKind, props.duration, props.period, setChartSrc]);
+
+  React.useEffect(() => {
+    addSubscription(
+      controllerContext.controller.attach().subscribe((_) => {
+        /* do nothing */
+      })
+    );
+  }, [addSubscription, controllerContext]);
+
+  const refresh = React.useCallback(() => {
+    controllerContext.controller.requestRefresh();
+  }, [controllerContext]);
+
+  React.useEffect(() => {
+    refresh();
+    addSubscription(interval(props.period * 1000).subscribe(() => refresh()));
+  }, [addSubscription, props.period, refresh]);
+
+  const popout = React.useCallback(() => {
+    if (chartSrc && dashboardUrl) {
+      window.open(chartSrc, '_blank');
+    }
+  }, [chartSrc, dashboardUrl]);
+
+  const cardStyle = React.useMemo(() => {
+    let height: number;
+    switch (props.chartKind) {
+      case 'Core Count':
+        height = 250;
+        break;
+      default:
+        height = 380;
+        break;
+    }
+    return hasRecording
+      ? {
+          height,
+        }
+      : {};
+  }, [hasRecording, props.chartKind]);
+
+  const resyncButton = React.useMemo(() => {
+    return (
+      <Button
+        key={0}
+        aria-label={t('CHART_CARD.BUTTONS.SYNC.LABEL', { chartKind: props.chartKind })}
+        onClick={updateKey}
+        variant="plain"
+        icon={<SyncAltIcon />}
+        isDisabled={!chartSrc || !dashboardUrl}
+      />
+    );
+  }, [t, props.chartKind, updateKey, chartSrc, dashboardUrl]);
+
+  const popoutButton = React.useMemo(() => {
+    return (
+      <Button
+        key={1}
+        aria-label={t('CHART_CARD.BUTTONS.POPOUT.LABEL', { chartKind: props.chartKind })}
+        onClick={popout}
+        variant="plain"
+        icon={<ExternalLinkAltIcon />}
+        isDisabled={!chartSrc || !dashboardUrl}
+      />
+    );
+  }, [t, props.chartKind, popout, chartSrc, dashboardUrl]);
+
+  const actions = React.useMemo(() => {
+    const a = props.actions || [];
+    return [resyncButton, popoutButton, ...a];
+  }, [props.actions, resyncButton, popoutButton]);
+
+  const header = React.useMemo(() => {
+    const isWide = props.span > 4;
+    const style = {
+      marginBottom: isWide ? '-2em' : '',
+    };
+    if (hasRecording) {
+      return (
+        <>
```

Comment:
Remove Fragment here? we probably wont have multiple headers but just for safety on the key issue.